v2v-helper/migrate/migrate.go

```diff
@@ -435,6 +457,98 @@ func (migobj *Migrate) CreateTargetInstance(vminfo vm.VMInfo) error {
 		return fmt.Errorf("failed to create VM: %s", err)
 	}
 	migobj.logMessage(fmt.Sprintf("VM created successfully: ID: %s", newVM.ID))
+
+	err = migobj.HealthCheck(ipaddresses)
+	if err != nil {
+		migobj.logMessage(fmt.Sprintf("Health Check failed: %s", err))
+	}
+
+	return nil
+}
+
+func (migobj *Migrate) pingVM(ips []string) error {
+	for _, ip := range ips {
+		migobj.logMessage(fmt.Sprintf("Pinging VM: %s", ip))
+		pinger, err := probing.NewPinger(ip)
+		if err != nil {
+			return fmt.Errorf("failed to create pinger: %s", err)
+		}
+		pinger.Count = 1
+		pinger.Timeout = time.Second * 10
+		err = pinger.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run pinger: %s", err)
+		}
+		if pinger.Statistics().PacketLoss == 0 {
+			migobj.logMessage("Ping succeeded")
+		} else {
+			return fmt.Errorf("Ping failed")
+		}
+	}
+	return nil
+}
+
+func (migobj *Migrate) checkHTTPGet(ips []string) error {
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+		Timeout: time.Second * 10,
+	}
+	for _, ip := range ips {
+		url := fmt.Sprintf("https://%s", ip)
```

Comment:
Hmm, even in this day and age, I think there will exist some legacy applications that are only able to handle `http`. Additionally, there might be other ports, eg. `8080`.

Reply:
Fixed